rust-sdk/core/src/quote/liquidity.rs

```diff
@@ -0,0 +1,1212 @@
+        tick_lower_index,
+        tick_upper_index,
+        transfer_fee_a,
+        transfer_fee_b,
+    )
+}
+
+/// Calculate the quote for increasing liquidity given a token b amount
+///
+/// # Parameters
+/// - `token_amount_b` - The amount of token b to increase
+/// - `slippage_tolerance` - The slippage tolerance in bps
+/// - `current_sqrt_price` - The current sqrt price of the pool
+/// - `tick_lower_index` - The lower tick index of the position
+/// - `tick_upper_index` - The upper tick index of the position
+/// - `transfer_fee_a` - The transfer fee for token A in bps
+/// - `transfer_fee_b` - The transfer fee for token B in bps
+///
+/// # Returns
+/// - An IncreaseLiquidityQuote struct containing the estimated token amounts
+#[cfg_attr(feature = "wasm", wasm_expose)]
+pub fn increase_liquidity_quote_b(
+    token_amount_b: u64,
+    slippage_tolerance_bps: u16,
+    current_sqrt_price: U128,
+    tick_lower_index: i32,
+    tick_upper_index: i32,
+    transfer_fee_a: Option<TransferFee>,
+    transfer_fee_b: Option<TransferFee>,
+) -> Result<IncreaseLiquidityQuote, ErrorCode> {
+    let tick_range = order_tick_indexes(tick_lower_index, tick_upper_index);
+    let token_delta_b = try_adjust_amount(token_amount_b, transfer_fee_b.into(), false)?;
+
+    if token_delta_b == 0 {
+        return Ok(IncreaseLiquidityQuote::default());
+    }
+
+    let current_sqrt_price: u128 = current_sqrt_price.into();
+    let sqrt_price_lower: u128 = tick_index_to_sqrt_price(tick_range.tick_lower_index).into();
+    let sqrt_price_upper: u128 = tick_index_to_sqrt_price(tick_range.tick_upper_index).into();
+
+    let position_status = position_status(
+        current_sqrt_price.into(),
+        tick_range.tick_lower_index,
+        tick_range.tick_upper_index,
+    );
+
+    let liquidity: u128 = match position_status {
+        PositionStatus::Invalid | PositionStatus::PriceBelowRange => 0,
+        PositionStatus::PriceAboveRange => {
+            try_get_liquidity_from_b(token_delta_b, sqrt_price_lower, sqrt_price_upper)?
+        }
+        PositionStatus::PriceInRange => {
+            try_get_liquidity_from_b(token_delta_b, sqrt_price_lower, current_sqrt_price)?
+        }
+    };
+
+    increase_liquidity_quote(
+        liquidity.into(),
+        slippage_tolerance_bps,
+        current_sqrt_price.into(),
+        tick_lower_index,
+        tick_upper_index,
+        transfer_fee_a,
+        transfer_fee_b,
+    )
+}
+
+// Private functions
+
+fn try_get_liquidity_from_a(
+    token_delta_a: u64,
+    sqrt_price_lower: u128,
+    sqrt_price_upper: u128,
+) -> Result<u128, ErrorCode> {
+    let sqrt_price_diff = sqrt_price_upper - sqrt_price_lower;
+    let result: U256 = <U256>::from(token_delta_a)
+        .wrapping_mul(sqrt_price_lower.into())
+        .wrapping_mul(sqrt_price_upper.into())
+        .wrapping_div(sqrt_price_diff.into())
+        .wrapping_shr(64);
+    result.try_into().map_err(|_| AMOUNT_EXCEEDS_MAX_U64)
```

Comment:
Could we remove `wrapping` here ?
I'm not sure why we need to use `wrapping` for div and shr.